### DISCUSSION TITLE
How to create bin files?

### DISCUSSION THREAD

**Discussion by tenprint** [drmortalwombat/oscar64#65] [Q&A] [answered]
I see there is a flag -tg=bin and I was able to compile a .c file to a .bin file but it didn't create what I was hoping to create.

What I'm looking to do is turn a series of const literals into .bin files so that I can load them when needed at runtime. Here's an small example:

`
// Create as level0.bin
static const Level level0 = {
    0, 10, "Foo", "Bar",  { 10, 100, 50, 25,  ... }
}
`

**Comment by drmortalwombat** [comment, marked as answer]
You need some boilerplate to avoid the runtime:

```
#pragma section( startup, 0)
#pragma region( startup, 0xc000, 0xd000, , , {startup})
#pragma region( main, 0xc000, 0xd000, , , {code, data, bss, heap, stack} )
#pragma heapsize(0)
#pragma stacksize(0)
__asm startup
{}
#pragma startup(startup)


__export char data[128] = {1, 2, 3, 4, 5, 6};

```

You can then compile without runtime: 

`oscar64 -tf=bin -rt= raw.c`

An alternative solution would be to use overlays:

```
#pragma overlay( ovl1, 1 )
#pragma section( bcode1, 0 )
#pragma section( bdata1, 0 )
#pragma region(bank1, 0xc000, 0xd000, , 1, { bcode1, bdata1 } )

#pragma data ( bdata1 )
__export char data[128] = {1, 2, 3, 4, 5, 6};
#pragma data ( data )

int main(void)
{
	return 0;
}

```

**Reply by tenprint** [reply]
Awesome, option 1 worked perfectly. Thanks for the fast response. BTW, enjoying Oscar64!
